lib/Ix/DBIC/ResultSet.pm

```diff
@@ -1030,7 +1038,7 @@ sub ix_get_list_updates ($self, $ctx, $arg = {}) {
 
     unless ($is_removed) {
       FILTER: for my $filter (keys $arg->{filter}->%*) {
-        if ($entity->$filter ne $arg->{filter}{$filter}) {
+        if (($entity->$filter // '') ne ($arg->{filter}{$filter} // '')) {
```

Comment:
This is going to treat `undef` as equivalent to `""`.  Should be fixed or commented.